### PR TITLE
Restructure docs

### PR DESCRIPTION
**Pull request by pellared**
Addresses https://github.com/signalfx/splunk-otel-go/pull/731#discussion_r843795147

Preview mode: https://github.com/pellared/splunk-otel-go/tree/restructure-docs

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,6 +77,9 @@ development documentation is hosted on this repository.
 You can send feedback about Splunk Observability docs by clicking the
 "Feedback" button on any of our documentation pages on the official docs site.
 
+The repository's "unofficial" documentation is kept under [docs](./docs)
```
Comment (theletterf) [inline CONTRIBUTING.md:80]:
```suggestion
The draft documentation of the repository is under [docs](./docs).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -77,6 +77,9 @@ development documentation is hosted on this repository.
...
+directory.
```
Comment (theletterf) [inline CONTRIBUTING.md:81]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -11,148 +11,22 @@
 
 The Splunk distribution of [OpenTelemetry
 Go](https://github.com/open-telemetry/opentelemetry-go) provides
-multiple packages that automatically instruments your Go
+multiple packages that instruments your Go
```
Comment (theletterf) [inline README.md:14]:
```suggestion
multiple packages that instrument your Go
```

File: README.md [hunk 4]
```diff
@@ -11,148 +11,22 @@
...
-[here](https://opentelemetry.io/docs/go/getting-started/).
+Read the official documentation for this distribution in the
+[Splunk Docs site](https://docs.splunk.com/Observability/gdi/get-data-in/application/go/get-started.html).
 
-## Troubleshooting
+## Contributing
 
-For troubleshooting information, see the
-[Troubleshooting](./docs/troubleshooting.md) documentation.
+Please read [CONTRIBUTING.md](CONTRIBUTING.md)
```
Comment (theletterf) [inline README.md:28]:
```suggestion
Read [CONTRIBUTING.md](CONTRIBUTING.md)
```

File: docs/README.md [hunk 5]
```diff
@@ -1,5 +1,128 @@
-# Documentation for the Splunk Distribution of OpenTelemetry Go
+# Unofficial documentation
```
Comment (theletterf) [inline docs/README.md:1]:
```suggestion
# Draft documentation
```

Comment (theletterf) [reply docs/README.md:1]: @pellared We don't give a special name to GH docs in other repos, but it's interesting to think of one. I'd rather go with "Draft", as "Unofficial" gives the idea that this is not Splunk's effort, while it is. We do support these docs, too.

**Review by theletterf — CHANGES_REQUESTED** [review]

**Review by theletterf — APPROVED** [review]

**Comment by pellared** [comment]
@MrAlias PTAL

**Comment by pellared** [comment]
> The RELEASING.md file was moved to the docs directory. It is developer documentation for the project not intended for the user. It should remain in the top level.

How about moving it to `docs/internal` as we did with @theletterf blessing [here](https://github.com/signalfx/signalfx-dotnet-tracing/tree/main/docs/internal) and [here](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/tree/main/docs)?

**Review by MrAlias — APPROVED** [review]

**Pull request merged**